### ISSUE TITLE
Messages in party chat is getting picked up by DiscordSRV

### ISSUE THREAD

**Issue by GrimPlayzs** [AlessioDP/Parties#241]
Currently on a survival server of mine players use the party chat feature to talk privately to a group of players however, my discord bot which relays messages to a discord channel picks up these chats. I should mention I use HuskChat for handling messages with SignedVelocity.

**Comment by AlessioDP** [comment]
How are you using the party chat feature? Command /p or toggling with /party chat then send messages to the chat?

**Comment by GrimPlayzs** [comment]
I misworded it by mistake, I meant that we type in /party chat and then the message after the command is entered. I use HuskChat version 2.7.1, And Parties version 3.2.9.

**Comment by AlessioDP** [comment]
SignedVelocity has been mentioned so I was supposing you were using Parties in Velocity, but you said that messages are sent to Discord via DiscordSRV (issue title). Is this discord message handled by DiscordSRV or HuskChat? (If DiscordSRV, which version?)

How did you setup Parties? Velocity or Bukkit only? I need more info about the server itself because I must understand what event is not getting cancelled correctly when my plugin handles the chat message.

PS. Parties is outdated

**Comment by GrimPlayzs** [comment]
Ever since I updated to SignedVelocity from UnsignVelocity messages began to be relayed to DiscordSRV with HuskChat as before it didn't with UnsignVelocity. As far I can tell the discord messages are being handled by DiscordSRV, but since all the chats are being handled by HustChat (through the proxy) I am not sure how DiscordSRV is getting the messages. To answer your other question Parties is only installed on Velocity. 

I am using version 1.26.2 of DiscordSRV.

**Comment by AlessioDP** [comment]
Did you install SignedVelocity in Bukkit servers?

**Comment by GrimPlayzs** [comment]
Yes I have all backend servers have SignedVelocity.

**Comment by AlessioDP** [comment]
I do not know, try to update all plugins. I tested it and if SignedVelocity is correctly installed, it works. For example, If I do not install it in sub servers, it won't work.

The message is correctly cancelled in subservers.

You may try to debug it by adding ` -Dio.github._4drian3d.signedvelocity.debug=true` at your server startup script (after java) in subservers, then post a full server log here after you sent a message from Velocity with /party chat toggled.

**Comment by GrimPlayzs** [comment]
Okay, I will try that. Won't be able to do it for a few hours.

**Comment by GrimPlayzs** [comment]
I noticed that the messages only appears for other players not me for example when I type something in my party chat it doesn't show up in that discord channel but every other user's messages does.

**Comment by AlessioDP** [comment]
Are you talking about chat messages or party chat messages?

**Comment by GrimPlayzs** [comment]
Party chat messages.

**Comment by AlessioDP** [comment]
Can you post server logs with debug enabled in Parties and DiscordSRV please

**Comment by GrimPlayzs** [comment]
The whole server log? Or just the log about the group chat message?

**Comment by AlessioDP** [comment]
Before and after you send a message in party chat and its triggered in Discord (both proxy and sub server). Entire server log is not a problem, just write the timestamp too of that moment.

**Comment by GrimPlayzs** [comment]
I also noticed the message doesn't appear in Discord if anyone not just me does /p (message). But if you first toggle into party chat by doing "/party chat" the message will always appear in Discord.

This is the log when I do "/p (message):

[11:27:03 INFO]: [SignedVelocity] [DEBUG] [Plugin Message] Received Valid Message
[11:27:03 INFO]: [SignedVelocity] [DEBUG] | Queue: COMMAND_RESULT
[11:27:03 INFO]: [SignedVelocity] [DEBUG] | Result: ALLOWED
[11:27:03 INFO]: [SignedVelocity] [DEBUG] | Message: null

Log when I first toggle into party chat by doing "/party chat" and then a message:

[11:28:55 INFO]: [SignedVelocity] [DEBUG] [Plugin Message] Received Valid Message
[11:28:55 INFO]: [SignedVelocity] [DEBUG] | Queue: CHAT_RESULT
[11:28:55 INFO]: [SignedVelocity] [DEBUG] | Result: CANCEL
[11:28:55 INFO]: [SignedVelocity] [DEBUG] | Message: null
[11:28:55 INFO]: [SignedVelocity] [DEBUG] [CHAT] Init Message Handling
[11:28:55 INFO]: [SignedVelocity] [DEBUG] [CHAT] Queueing Next Result
[11:28:55 INFO]: [SignedVelocity] [DEBUG] [CHAT] Future Done: true
[11:28:55 INFO]: [SignedVelocity] [DEBUG] [CHAT] Waiting for next result
[11:28:55 INFO]: [SignedVelocity] [DEBUG] [CHAT] Next Result
[11:28:55 INFO]: [SignedVelocity] [DEBUG] [CHAT] Cancelled Message.
[11:28:55 INFO]: [SignedVelocity] [DEBUG] Original Message: Test
[11:28:55 INFO]: [SignedVelocity] [DEBUG] [CHAT] Result applied

**Comment by AlessioDP** [comment]
I cannot understand it correctly without the server log but just those messages, but the problem is that this is not an issue of Parties but something else that is not handling events correctly.

Parties in velocity **cancels** the event and with SignedVelocity is sent the canceled message to sub servers. If SignedVelocity is correctly setup, it handles the message in sub servers and cancel that event too. If SignedVelocity does not work correctly in sub server, it happens what you said.

**Comment by GrimPlayzs** [comment]
You said in your test that you didn't have this issue. Could another plugin I have installed be causing this? The only other plugin I have on that sub-server that does something with the chat is FreedomChat.

**Comment by AlessioDP** [comment]
It may be, if a plugin does not check that an event is cancelled before handle the message, its an issue.

**Comment by GrimPlayzs** [comment]
No, it is not that plugin. I started the server without that plugin and it still happens.

**Comment by AlessioDP** [comment]
Send full both server logs please with you/someone that executes a party message with chat (/party chat on).

**Comment by GrimPlayzs** [comment]
Alright will do.
